### ISSUE TITLE
`captureNames` cannot be used to map group indexes to group names

### ISSUE THREAD

**Issue by RunDevelopment** [2fd/rregex#40]
Hi! I just tried out the new capture APIs from v1.8.0 and founds that `RRegex#captureNames()` is not as useful as is could be. Example:

```js
> const { RRegex } = require("rregex")
undefined
> const r = new RRegex("(a)(?P<foo>b)")
undefined
> r.capturesLength()
3
> r.captureNames()
[ 'foo' ]
```

In the above example, we cannot say *which* capturing group has the name `foo`. It could be the first, it could be the second. This limits the usefulness of `captureNames`, because it cannot be used to map indexes to capture names and vise versa.

Note that [the rust regex API for `capture_names`](https://docs.rs/regex/latest/regex/struct.Regex.html#method.capture_names) does not have this issue, because it yields `None` for capture groups without a name. In JS terms, it would return `[ null, null, 'foo' ]`. This makes it trivial to map from group index to group name.

I would suggest copying the `capture_names` function and inserting`null` for groups without names. This would be a breaking change, but the current version of `captureNames` is not fully functional anyway, so one could see this as a bug fix.

**Comment by 2fd** [comment]
Hi @RunDevelopment! 

Thanks for pointing that out is already fixed on `1.10.1`, `1.9.1`, and `1.8.1`

**Comment by RunDevelopment** [comment]
Thank you!
